src/microbe_stage/SpawnSystem.cs

```diff
@@ -207,10 +207,12 @@ private void HandleQueuedSpawns(ref int spawnsLeftThisFrame)
         {
             var spawn = queuedSpawns.First();
             var enumerator = spawn.Spawns;
+            bool spawningClouds = spawn.SpawnType is CompoundCloudSpawner;
 
-            bool finished = false;
+            // Discard the whole spawn if we're too close to the player
+            bool finished = (playerPosition - spawn.Location).Length() < Constants.SPAWN_SECTOR_SIZE;
```

Comment:
I would prefer that too, but since the enumerator spawns each time it's iterated, there's no way to check that here before the thing gets spawned. It would have to be a check in the spawn method of every spawn type, which would require passing the player location to each of these spawn types and doing this comparison for every single spawn, not just the queued ones. At that point I feel it's just too much of a mess to be worth it.

Alternatively, it could despawn straight away after being spawned, but I can't see that ending well performance-wise, and that might even still be visible to the player. The other alternative is to overhaul how the spawn system enumerates stuff, which I'm quite against considering all the extra design and testing that would require.

Basically, I don't see any other way of doing this that's anywhere near this simple. Especially since as far as I can tell, location here will always be the correct location anyway. But I'm willing to believe I've missed something vital, so feel free to correct me.

Reply:
> I would prefer that too, but since the enumerator spawns each time it's iterated, there's no way to check that here before the thing gets spawned. It would have to be a check in the spawn method of every spawn type, which would require passing the player location to each of these spawn types and doing this comparison for every single spawn, not just the queued ones. At that point I feel it's just too much of a mess to be worth it.

I said earlier that it's *probably* not too much of a problem to just despawn the entity immediately.
Anything that happens during a single frame should not be able to be visible.

> location here will always be the correct location anyway.

Your approach anyway is not actually bulletproof either because the spawners don't use the exact position they are given, so depending on that to skip spawning has situations where someone might write a badly behaving spawner or something that causes the approach to break.